adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,36 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
+#include <algorithm>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string &source)
+{
+    std::stack<char> stack{};
+
+    for (const auto c : source)
+    {
+        const auto typeIt = std::find_if(brackets.cbegin(), brackets.cend(),
+                                         [c](const std::pair<char, char>& type)
+                                         { return c == type.first || c == type.second; });
+        if (brackets.cend() == typeIt)
+        {
+            continue;
+        }
+        else if (!stack.empty() && stack.top() == typeIt->first && c == typeIt->second)
```

Comment:
For explicitness, please name this conditions, because this expression code is hard to read now

Reply:
Fixed with lambda